grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -2626,12 +2619,13 @@ class StockMovementService {
             log.info "Shipment workflow " + shipmentWorkflow
             if (shipmentWorkflow) {
                 shipmentWorkflow.documentTemplates.each { Document documentTemplate ->
-                    def action = documentTemplate.documentType.documentCode == DocumentCode.INVOICE_TEMPLATE ? "download" : "render"
+                    def action = getActionByDocumentCode(documentTemplate.documentType?.documentCode)
+                    def isInvoiceTemplate = documentTemplate.documentType?.documentCode == DocumentCode.INVOICE_TEMPLATE
                     documentList << [
                             name        : documentTemplate?.name,
                             documentType: documentTemplate?.documentType?.name,
                             contentType : documentTemplate?.contentType,
-                            stepNumber  : null,
+                            stepNumber  : isInvoiceTemplate ? 5 : null,
```

Comment:
We need to rethink this stepNumber thing at some point. I'll take the blame for that.